atomic_reactor/util.py

```diff
@@ -985,8 +985,93 @@ def get_manifest_list(image, registry, insecure=False, dockercfg_path=None):
     return response
 
 
+def get_all_manifest_digests(image, registry, insecure=False, dockercfg_path=None,
+                             versions=('v1', 'v2', 'v2_list', 'oci', 'oci_index')):
+    """Return manifest digests for image.
+
+    :param image: ImageName, the remote image to inspect
+    :param registry: str, URI for registry, if URI schema is not provided,
+                          https:// will be used
+    :param insecure: bool, when True registry's cert is not verified
+    :param dockercfg_path: str, dirname of .dockercfg location
+
+    :return: dict of successful responses, with versions as keys
+    """
+    digests = {}
+    registry_session = RegistrySession(registry, insecure=insecure, dockercfg_path=dockercfg_path)
+    for version in versions:
+        response, _ = get_manifest(image, registry_session, version)
+        if response:
+            digests[version] = response
+
+    return digests
+
+
+def get_inspect_for_image(image, registry, insecure=False, dockercfg_path=None):
+    """Return inspect for image.
+
+    :param image: ImageName, the remote image to inspect
+    :param registry: str, URI for registry, if URI schema is not provided,
+                          https:// will be used
+    :param insecure: bool, when True registry's cert is not verified
+    :param dockercfg_path: str, dirname of .dockercfg location
+
+    :return: dict of inspected image
+    """
+    all_man_digests = get_all_manifest_digests(image, registry, insecure=insecure,
+                                               dockercfg_path=dockercfg_path)
+    blob_config = None
+    config_digest = None
+    image_inspect = {}
+
+    # we have manifest list (get digest for 1st platform)
+    if 'v2_list' in all_man_digests:
+        man_list_json = all_man_digests['v2_list'].json()
+        v2_digest = man_list_json['manifests'][0]['digest']
+        blob_config, config_digest = get_config_from_registry(image, registry, v2_digest,
+                                                              insecure=insecure, version='v2',
+                                                              return_config_digest=True,
+                                                              dockercfg_path=dockercfg_path)
+    # get config for v2 digest
+    elif 'v2' in all_man_digests:
+        blob_config, config_digest = get_config_from_registry(image, registry, image.tag,
+                                                              insecure=insecure, version='v2',
+                                                              return_config_digest=True,
+                                                              dockercfg_path=dockercfg_path)
+    # read config from v1
+    elif 'v1' in all_man_digests:
+        v1_json = all_man_digests['v1'].json()
+        blob_config = json.loads(v1_json['history'][0]['v1Compatibility'].decode('utf-8'))
+    else:
+        raise NotImplementedError("OCI isn't implemented")
```

Comment:
We don't really know this is an OCI image. Maybe something more generic like "No v2 schema 1 image, or v2 schema 2 image or list, found".

Reply:
ok